### PR TITLE
alternative to #22219; adds a pointer wrapper for T destructor

### PR DESCRIPTION
**Pull request by ringabout**
alternative to #22219

### CODE REVIEW

File: compiler/ccgtypes.nim [hunk 1]
```diff
@@ -1527,8 +1566,14 @@ proc genHook(m: BModule; t: PType; info: TLineInfo; op: TTypeAttachedOp; result:
       localError(m.config, info,
         theProc.name.s & " needs to have the 'nimcall' calling convention")
 
-    genProc(m, theProc)
-    result.add theProc.loc.r
+    if op == attachedDestructor and t.kind == tySequence:
```
Comment (Araq) [inline compiler/ccgtypes.nim]:
```suggestion
    if op == attachedDestructor:
```

Comment (ringabout) [reply compiler/ccgtypes.nim]: Not every type supports T destructors. I suppose that's going to cause problems for var T destructors. It should be fine for `ref string` and `ref object`.

File: compiler/ccgtypes.nim [hunk 2]
```diff
@@ -1527,8 +1566,14 @@ proc genHook(m: BModule; t: PType; info: TLineInfo; op: TTypeAttachedOp; result:
...
+      let wrapper = generateRttiDestructor(m.g.graph, t, theProc.owner, attachedDestructor,
```
Comment (ringabout) [inline compiler/ccgtypes.nim:1576]: Should I cache the generated symbols in the modulegraph?

File: compiler/ccgtypes.nim [hunk 3]
```diff
@@ -1517,6 +1517,46 @@ proc genTypeInfo2Name(m: BModule; t: PType): Rope =
 
 proc isTrivialProc(g: ModuleGraph; s: PSym): bool {.inline.} = getBody(g, s).len == 0
 
+proc makePtrType(baseType: PType; idgen: IdGenerator): PType =
+  result = newType(tyPtr, nextTypeId idgen, baseType.owner)
+  addSonSkipIntLit(result, baseType, idgen)
+
+proc generateRttiDestructor(g: ModuleGraph; typ: PType; owner: PSym; kind: TTypeAttachedOp;
+              info: TLineInfo; idgen: IdGenerator; theProc: PSym): PSym =
+  # the warpper is roughly like:
```
Comment (ringabout) [inline compiler/ccgtypes.nim]:
```suggestion
  # the wrapper is roughly like:
```

File: compiler/ccgtypes.nim (hunk 1 above)
Comment (Araq) [reply compiler/ccgtypes.nim]: That only means you need to do some introspection and see if the destructor takes a `var T` or not. But special casing `tySequence` is clearly wrong.

**Pull request merged**